File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.cpp
class Solution {
public:
    int longestSubsequence(vector<int>& nums) {
        int n = nums.size();
        int xorSum = 0;
        bool nonZero = false;
        for(int i = 0; i < n; i++)
        {
            xorSum ^= nums[i];
            if(nums[i])
            {
                nonZero = true;
            }
        }
        return xorSum > 0 ? n : nonZero ? n - 1 : 0;
    }
};

File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.py
class Solution:
    def longestSubsequence(self, nums: List[int]) -> int:
        n = len(nums)
        xorSum = 0
        nonZero = False
        for num in nums:
            xorSum ^= num
            if num:
                nonZero = True
        return n if xorSum > 0 else n - 1 if nonZero else 0